index.js

```diff
@@ -218,16 +221,49 @@ serveIndex.html = function _html(req, res, files, next, dir, showUp, icons, path
  * Respond with application/json.
  */
 
-serveIndex.json = function _json(req, res, files) {
-  send(res, 'application/json', JSON.stringify(files))
+serveIndex.json = function _json(req, res, dir, files) {
+  var directory = {
+    name: dir.name,
+    type: 'inode/directory',
+    size: dir.stat.size,
+    lastModified: dir.stat.mtime.toISOString()
+  }
+
+  // similar to W3 FileAPI Object
+  var nodes = files.map(function (file) {
+    var ext = extname(file.name)
+    var mimetype = mime.lookup(ext)
+    return {
+      name: file.name,
+      type: file.stat.isDirectory() ? 'inode/directory' : mimetype,
+      size: file.stat.size,
+      lastModified: file.stat.mtime.toISOString()
+    }
+  })
+  send(res, 'application/json', JSON.stringify({ directory: directory, nodes: nodes }))
 };
 
 /**
  * Respond with text/plain.
  */
 
-serveIndex.plain = function _plain(req, res, files) {
-  send(res, 'text/plain', (files.join('\n') + '\n'))
+serveIndex.plain = function _plain(req, res, dir, files) {
+  var directory = {
+    name: dir.name.replace(/\/$/, '\/')
```

Comment:
Does this replace actually do anything? I can't figure out what it does, can you ellaborate?

Reply:
change to `dir.name.replace(/\/?$/, '/')`
/ => /
demo => demo/